server/src-lib/Hasura/GraphQL/Resolve/Select.hs

```diff
@@ -44,6 +45,27 @@ withSelSet selSet f =
     res <- f fld
     return (G.unName $ G.unAlias $ _fAlias fld, res)
 
+jsonPathToColExp :: (MonadError QErr m) => T.Text -> m S.SQLExp
+jsonPathToColExp t = case parseJSONPath t of
+  Left s       -> throwVE $ T.pack s
+  Right jPaths -> return $ S.SEArray $ map elToColExp jPaths
+  where
+    elToColExp (Key k)   = S.SELit k
+    elToColExp (Index i) = S.SELit $ T.pack (show i)
+
+
+argsToColOp :: (MonadError QErr m) => ArgsMap -> m (Maybe RS.ColOp)
+argsToColOp args = case (Map.lookup "path" args) of
+  Nothing  -> return Nothing
+  Just val -> toOp val
+  where
+    toJsonPathExp t = jsonPathToColExp t
+      >>= (return . Just . (RS.ColOp S.jsonbPathOp))
+    toOp = \case
+      AGScalar _ (Just (PGValText t)) -> toJsonPathExp t
+      AGScalar _ (Just (PGValVarchar t)) -> toJsonPathExp t
+      _ -> return Nothing
```

Comment:
We're leaving values other than `PGValText` and `PGValVarchar`. So, in case of values other than `String` kinds we have to `500 Internal error`. Use `throw500`. However, GraphQL schema validation makes sure it will be `String`.

I suggest you write a function in `GraphQL/Resolve/InputValue.hs` module
```haskell
onlyText
  :: (MonadError QErr m)
  => PGColValue -> m Text
onlyText = \case
  PGValText t -> return t
  _           -> throw500 "expecting text for asPGColText"
asPGColTextM
  :: (MonadError QErr m)
  => AnnGValue -> m (Maybe Text)
asPGColTextM val = do
  pgColValM <- fmap snd <$> asPGColValM val
  mapM onlyText pgColValM
```
use `onlyText` function in `asPGColText` to replace pattern matching code.

Reply:
When I saw in `GraphQL/Resolve/InputValue.hs ` module, `asPGColText` did exist already. I think it can be reused.

https://github.com/hasura/graphql-engine/blob/f794653b69279f7b123353997e0428d00f9c7814/server/src-lib/Hasura/GraphQL/Resolve/InputValue.hs#L124

Reply:
`asPGColText` is used in case of `!String` graphQL type. Here, we're parsing `path: String`. Currently we're treating absence of a key and explicit `null` value to a key as two different things( refer https://github.com/hasura/graphql-engine/issues/1456 ). So, you'll need to use `asPGColTextM` in to handle `{path: null}`.